pkgs/development/libraries/flutter-engine/default.nix

```diff
@@ -0,0 +1,283 @@
+          '';
+        };
+      in
+      stdenvNoCC.mkDerivation rec {
+        pname = "flutter-engine-${runtimeMode}";
+        version = builtins.substring 0 7 args.version;
+        inherit enableParallelBuilding runtimeMode src;
+
+        nativeBuildInputs = [ git python3 pkg-config ninja stdenv.cc.libc ];
+
+        requiredSystemFeatures = optional enableParallelBuilding "big-parallel";
+
+        ## FIXME: create a sysroot toolchain for Nix and use it instead
+        ## TODO: generate for other platforms based on "targetPlatform"
+        configurePhase = ''
+          ./src/flutter/tools/gn \
+            --runtime-mode $runtimeMode \
+            --no-build-glfw-shell \
+            --prebuilt-dart-sdk \
+            --embedder-for-target \
+            --no-goma \
+            --out-dir $out/lib/flutter \
+            --target-dir $runtimeMode
+        '';
+
+        ## TODO: add more builds and possibly better cross-compiling
+        buildPhase = ''
+          cd $out/lib/flutter/out/$runtimeMode
+          export TERM=dumb
+          # Comes from "pkgs/development/tools/build-managers/ninja/setup-hook.sh"
+          local buildCores=1
+          if [ "''${enableParallelBuilding-1}" ]; then
+            buildCores="$NIX_BUILD_CORES"
+          fi
+          local flagsArray=(
+            -j$buildCores
+            $ninjaFlags "''${ninjaFlagsArray[@]}"
+          )
+          echo "Building flatc"
+          ninja flatc "''${flagsArray[@]}"
+          patchelf --set-interpreter ${stdenv.cc.libc}/lib/${interpreter} flatc
+          echo "Building blobcat"
+          ninja blobcat "''${flagsArray[@]}"
+          patchelf --set-interpreter ${stdenv.cc.libc}/lib/${interpreter} blobcat
+          echo "Building gen_snapshot"
+          ninja gen_snapshot "''${flagsArray[@]}"
+          patchelf --set-interpreter ${stdenv.cc.libc}/lib/${interpreter} gen_snapshot
+          echo "Building impellerc"
+          ninja impellerc "''${flagsArray[@]}"
+          patchelf --set-interpreter ${stdenv.cc.libc}/lib/${interpreter} impellerc
+          echo "Building Flutter Engine library"
+          ninja flutter_engine_library "''${flagsArray[@]}"
+          echo "Building icudtl.dat"
+          ninja icudtl.dat "''${flagsArray[@]}"
+          ninja flutter_embedder.h "''${flagsArray[@]}"
+        '';
+
+        installPhase = ''
+          mkdir -p $out/lib/pkgconfig
+          substituteAll ${./flutter-engine.pc} $out/lib/pkgconfig/flutter-engine-$runtimeMode.pc
+        '';
+
+        meta = {
+          description = "The engine for Flutter (${runtimeMode} mode)";
+          homepage = "https://github.com/flutter/engine";
+          license = licenses.bsd3;
+          maintainers = with maintainers; [ RossComputerGuy ];
+
+          # This build takes a long time; particularly on slow architectures
+          # TODO update times
+          # 1 hour on 32x3.6GHz -> maybe 12 hours on 4x2.4GHz
+          timeout = 24 * 3600;
```

Comment:
update times. currently like 1/3 of that

Reply:
Done